src/common/com/intellij/plugins/haxe/model/HaxeClassModel.java

```diff
@@ -102,9 +102,17 @@ public boolean isAbstract() {
   public HaxeTypeOrAnonymous getAbstractUnderlyingType() {
     if (!isAbstract()) return null;
     PsiElement[] children = getPsi().getChildren();
-    if (children.length >= 2) {
-      if (children[1] instanceof HaxeTypeOrAnonymous) {
-        return (HaxeTypeOrAnonymous)children[1];
+    // FIX: support list of metas before ComponentName
+    for(int i = 0; i < children.length; ++i) {
+      PsiElement child = children[i];
+      if(child instanceof HaxeComponentName) {
+        if(i + 1 < children.length) {
+          child = children[i + 1];
+          if(child instanceof HaxeTypeOrAnonymous) {
+            return (HaxeTypeOrAnonymous)child;
+          }
```

Comment:
In previous logic Semantic model stops resolving the abstract underlying type if before abstract keyword was metas. *NOTE: This logic is used to resolve "this." completion for abstract types*
So, for `abstract MyInt(Int)` previous logic works.
But if we are using `@:enum abstract MyEnum(Int)` - here `children[1]` is not `HaxeTypeOrAnonymous` anymore, it's abstract keyword:
![image](https://cloud.githubusercontent.com/assets/3038174/13300064/70793b1e-db4f-11e5-8c55-2179701b2eb8.png)

To fix this incorrect behavior and not change too much the current approach getting this from psi children (looked at `// TODO` on top) - I've decided to just search first TypeOrAnonymous element following by the first NameComponent (which is abstract name identifier).